src/hooks/core/push-notification/usePushNotificationHook.ts

```diff
@@ -0,0 +1,218 @@
+      id: '123',
+      title,
+      channelId: 'bettersosialid',
+      message: body,
+      data: remoteMessage.data
+    });
+  };
+
+  const __handlePushNotif = (remoteMessage) => {
+    const {data} = remoteMessage;
+    if (data.channel_type !== 3) {
+      if (isIos) {
+        __pushNotifIos(remoteMessage);
+      } else {
+        __pushNotifAndroid(remoteMessage);
+      }
+    }
+  };
+
+  const __updateProfileAtomId = async () => {
+    const signedUserId = await getUserId();
+    const anonUserId = await getAnonymousUserId();
+    const token = TokenStorage.get(ITokenEnum.token);
+    const anonymousToken = TokenStorage.get(ITokenEnum.anonymousToken);
+
+    setProfileAtom({
+      anonProfileId: anonUserId,
+      signedProfileId: signedUserId,
+      token,
+      anonymousToken
+    });
+  };
+  let isOpenNotification = false;
+
+  const helperNavigationResetWithData = (screenData) => {
+    setTimeout(() => {
+      if (!isOpenNotification) {
+        isOpenNotification = true;
+        const routes = [
+          {
+            name: 'AuthenticatedStack',
+            params: {
+              screen: 'HomeTabs',
+              params: {
+                screen: 'SignedChannelList'
+              }
+            }
+          }
+        ];
+
+        if (screenData)
+          routes.push({
+            name: 'AuthenticatedStack',
+            params: {
+              ...screenData
+            }
+          });
+
+        navigation.reset({
+          index: screenData ? 2 : 1,
+          routes
+        });
+      }
+    }, 500);
+  };
+
+  const __handleNotification = async (notification) => {
+    if (notification.data.type === 'feed' || notification.data.type === 'reaction') {
+      navigation.navigate('PostDetailPage', {
+        feedId: notification.data.feed_id,
+        is_from_pn: true
+      });
+    }
+    if (notification.data.type === 'follow_user') {
+      navigation.navigate('OtherProfile', {
+        data: {
+          user_id: notification.data.user_id,
+          other_id: notification.data.user_id_follower,
+          username: notification.data.username_follower
+        }
+      });
+    }
+    if (notification.data.type === 'message.new') {
+      if (notification?.data?.receiver_id === signedProfileId) {
+        const channel = new ChannelList({
+          id: notification?.data?.channel_id,
+          channelType: 'PM'
+        });
+        setIsLoadingFetchingChannelDetail(true);
+        try {
+          await fetchChannelDetail(channel);
+          const selectedChannel = await ChannelList.getSchemaById(
+            localDb,
+            notification?.data?.channel_id
+          );
+          setSelectedChannel(selectedChannel);
+          helperNavigationResetWithData({
+            screen: 'SignedChatScreen'
+          });
+        } catch (e) {
+          console.log('error', e);
+        } finally {
+          setIsLoadingFetchingChannelDetail(false);
+        }
+      }
+    }
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The function `__handleNotification` handles different types of notifications and navigates to the appropriate screen based on the notification type. This function is well implemented, but it could be improved by separating the logic for handling different types of notifications into separate functions. This would make the code more modular and easier to maintain.



<!-- This is an auto-generated comment by OSS CodeRabbit -->